x-pack/test/detection_engine_api_integration/utils.ts

```diff
@@ -457,3 +473,29 @@ export const getComplexRuleOutput = (ruleId = 'rule-1'): Partial<RulesSchema> =>
   query: 'user.name: root or user.name: admin',
   exceptions_list: [],
 });
+
+// Similar to ReactJs's waitFor from here: https://testing-library.com/docs/dom-testing-library/api-async#waitfor
+export const waitFor = async (
+  functionToTest: () => Promise<boolean>,
+  maxTimeout: number = 5000,
+  timeoutWait: number = 10
+) => {
```

Comment:
And 🙌  for the new util here -- will be nice to easily leverage this functionality in the integration tests as well now too.